### PR TITLE
eval-adxl355-pmdz: Add support for new trigger structure

### PR DESCRIPTION
**Pull request by rbolboac**
- Adapt adxl355 driver and eval-adxl355-pmdz project for new trigger structure.
- Add IIO_TRIGGER_EXAMPLE support for maxim platform
- Minor code refactoring

### CODE REVIEW

**Comment by rbolboac** [comment]
Can be merged only after https://github.com/analogdevicesinc/no-OS/pull/1363 is merged.

**Comment by rbolboac** [comment]
- master rebase

**Comment by rbolboac** [comment]
- master rebasse

File: projects/eval-adxl355-pmdz/src/platform/stm32/platform_src.mk [hunk 1]
```diff
@@ -1,14 +1,14 @@
 INCS += $(PLATFORM_DRIVERS)/stm32_delay.h     \
-		$(PLATFORM_DRIVERS)/stm32_gpio.h      \
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/stm32/platform_src.mk:2]: as far as i can see in other projects, the *.mk files are using tabs for indentation. so not sure if this change is required.

File: projects/eval-adxl355-pmdz/src/common/common_data.c [hunk 2]
```diff
@@ -46,44 +46,48 @@
 /********************** Macros and Constants Definitions **********************/
 /******************************************************************************/
 #ifdef DUMMY_EXAMPLE
-struct no_os_uart_init_param uip = {
+struct no_os_uart_init_param adxl355_uart_ip = {
 	.device_id = UART_DEVICE_ID,
-	.baud_rate = 115200,
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/common/common_data.c:51]: is this change related to the commit description? looks like this commit does more than adapting to the new trigger structure.

**Review by amiclaus — COMMENTED** [review]
some minor comments. Although, I think the adapt to new trigger struct commits are not quite doing only that. Moreover, the changes in the driver and project with respect to this subject (new trigger structure) are interdependent, therefore they should be in the same commit, otherwise the driver changes commit will break the project at that specific point in commit history.

File: projects/eval-adxl355-pmdz/src/common/common_data.c (hunk 2 above)
Comment (rbolboac) [reply projects/eval-adxl355-pmdz/src/common/common_data.c:51]: no, this change should go to the code refactoring part. I will update this. thanks!

**Comment by rbolboac** [comment]
> some minor comments. Although, I think the adapt to new trigger struct commits are not quite doing only that. Moreover, the changes in the driver and project with respect to this subject (new trigger structure) are interdependent, therefore they should be in the same commit, otherwise the driver changes commit will break the project at that specific point in commit history.

I fixed the changes related to the new trigger structured and I added all of them in a single commit such that the build is successful for each commit and the project is working as expected.

**Review by buha — APPROVED** [review]
looks good to me

**Review by amiclaus — APPROVED** [review]
